### PR TITLE
ARROW-17211: [Java] Fix java-jar nightly on gh & self-hosted runners

### PR DESCRIPTION
**Pull request by assignUser**

### CODE REVIEW

**Comment by assignUser** [comment]
@github-actions crossbow submit java-jars

File: ci/scripts/java_full_build.sh [hunk 1]
```diff
@@ -28,10 +28,13 @@ pushd ${arrow_dir}/java
 
 # Ensure that there is no old jar
 # inside the maven repository
-find ~/.m2/repository/org/apache/arrow \
+maven_repo=~/.m2/repository/org/apache/arrow
+if [ -d $maven_repo ]; then
+find $maven_repo \
```
Comment (kou) [inline ci/scripts/java_full_build.sh:33]: Could you indent this?

**Review by kou — APPROVED** [review]
+1

**Review by raulcd — APPROVED** [review]
Thanks!

**Pull request merged**